### PR TITLE
Three.js graphic demo

### PR DESCRIPTION
**Pull request by jacklehamster**
Updated graphic engine to use THREE.js

Demo at graphic_3d_test.html

### CODE REVIEW

File: scripts/graphic-test-3d.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+requirejs.config({
+//    enforceDefine: true,
```
Comment (Wonay) [inline scripts/graphic-test-3d.js]: You can start turning those to true now that we set everything up.

File: scripts/graphics/main-3d.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+define(
+    [
+        './actiongenerator.js',
+        './viewmodel.js',
+        './threejsviewer.js',
+    ],
+    function(ActionGenerator, ViewModel, THREEViewer) {
+        var canvas = document.getElementById("canvas");
+
+        var generator = new ActionGenerator({
```
Comment (Wonay) [inline scripts/graphics/main-3d.js]: There is the Location class to have a standard tool for location and distance

Comment (Wonay) [inline scripts/graphics/main-3d.js]: You can extend it for 3 dimension and have (z = 0 ) so it still works in 2d

**Comment by Wonay** [comment]
Can you merged once it's approved or am I the only one who can ?

**Comment by Wonay** [comment]
Approved ! 

Can you merge or do I need to do it ? Just curious.

File: scripts/graphics/main-3d.js (hunk 2 above)
Comment (jacklehamster) [inline scripts/graphics/main-3d.js]: I'll get to it later. Just wanted to get this working in THREE.js first

Comment (Wonay) [inline scripts/graphics/main-3d.js]: okay !

**Pull request merged**